### PR TITLE
Support injecting OIDC WireMockServer into tests

### PR DESCRIPTION
**Pull request by dimas-b**
* Define a custom annotation: InjectWireMock

* Override inject(TestInjector) in OidcWiremockTestResource
to allow tests to receive the WireMockServer
used by OidcWiremockTestResource

This is to allow (external) tests to mock
extra URLs, that are not covered by OidcWiremockTestResource

----

If this PR is accepted, please consider back-porting to Quarkus 2.2.x

### CODE REVIEW

**Comment by sberyozkin** [comment]
Hi @dimas-b, thanks for this PR - does the test pass with `-Dnative` ? Also, perhaps the existing oidc client or oidc test can be enhanced a little bit to show for the users to see how it can be practically applied and for it to be documented as well ?

@stuartwdouglas Can you also have a look please ? May be instantiating `TestManager` can be avoid somehow ?

Thanks

**Comment by dimas-b** [comment]
> does the test pass with -Dnative ?

It does, although I do not think this flag makes a real difference since the test does not actually bring a full Quarkus server up.

**Comment by dimas-b** [comment]
@sberyozkin :
> perhaps the existing oidc client or oidc test can be enhanced a little bit to show for the users to see how it can be practically applied and for it to be documented as well ?

I've pushed another commit to add a test with an invalid toke to `BearerTokenAuthorizationTest` : 0ac955da351d97f3ef55438b01f82d037bfb91ba. This is the use case that prompted me to open this PR, actually.

Is this what you meant for providing a practical example?

**Review by stuartwdouglas — APPROVED** [review]

**Comment by sberyozkin** [comment]
@dimas-b thanks, looks good with a new test. Can you please squash the commits ?

One other question, do you think it can make sense to drop `Inject` from `InjectWireMock` since it is implied ? I don't really mind if it stays as `InjectWireMock` but I'm not sure we have any other marker annotation in Quarkus with the `Inject`in its name  indicating the target has to be injected. I'll let you decide if you'd like to keep it or not...

Thanks

**Comment by dimas-b** [comment]
@sberyozkin : There's already a `io.quarkus.it.hibernate.search.elasticsearch.aws.InjectWireMock` for a similar use case, so I followed that pattern. I'm ok to rename though. How about `OidcWireMock` since it works in concert with `OidcWiremockTestResource`.

I'll squash as soon as we settle on a name for the annotation :)

**Comment by sberyozkin** [comment]
@dimas-b Sorry, missed your last comment, I see, following the same pattern if it is already used in one other place in Quarkus makes sense, though I also like your idea of naming it as `OidcWireMock` as the wire mock itself is offered by the test resource factory with its name starting with `Oidc` - so if it works for you then please rename and squash and we'll merge, thanks

**Comment by dimas-b** [comment]
Renamed, squashed and rebased

**Review by sberyozkin — APPROVED** [review]
@dimas-b thanks for your PR. I'll follow up with a doc update as I don't want to hold this PR any longer. (if you can find some time - please do a small example to https://quarkus.io/guides/security-openid-connect#integration-testing-wiremock to introduce this option)

**Comment by dimas-b** [comment]
Follow-up docs PR: #20118

**Comment by dimas-b** [comment]
@sberyozkin : Would it be possible to backport this to 2.2.x?

**Pull request merged**
